src/helper/remote_config/asm_features_listener.cpp

```diff
@@ -34,15 +34,27 @@ void dds::remote_config::asm_features_listener::on_update(const config &config)
     }
 
     auto enabled_itr = asm_itr->value.FindMember("enabled");
-    if (enabled_itr == asm_itr->value.MemberEnd() ||
-        enabled_itr->value.GetType() != rapidjson::kStringType) {
+    if (enabled_itr == asm_itr->value.MemberEnd()) {
         throw error_applying_config(
             "Invalid config json encoded contents: enabled key missing");
     }
 
-    if (strcmp("true", enabled_itr->value.GetString()) == 0) {
+    if (enabled_itr->value.GetType() == rapidjson::kStringType) {
+        if (strcmp("true", enabled_itr->value.GetString()) == 0) {
```

Comment:
```suggestion
        if (enabled_itr->value.GetString() == "true") {
```

Reply:
Also is it possible for this value to be upper case?

Reply:
Done